### PR TITLE
Increase test coverage

### PR DESCRIPTION
**Pull request by lzpap**
Introduce new test methods:
 - `test_bundle_is_valid`
 - `test_wrongly_ordered_bundle`
 - `test_wrong_index`
 - `test_wrong_hash`
 - `test_timestamp`

Also, restructured `CcurlPowTestCase` to reduce the number of pow calculations and therefore shorten test run times.

Add hash validation to `attach_to_tangle`. Refactor function to recalculate pow until transaction hash becomes valid. See [commit](https://github.com/iotaledger/ccurl.interface.py/pull/11/commits/fbed75355ab6dca31642607ef15a080bbbd8430c) for more info.

All ideas for new tests are welcome!

### CODE REVIEW

File: pow/ccurl_interface.py [hunk 1]
```diff
@@ -111,41 +111,78 @@ def attach_to_tangle(bundle_trytes, # Iterable[TryteString]
...
-
-        # Create powed txn object
-        powed_txn = Transaction.from_tryte_string(
-            trytes=powed_txn_trytes,
-            hash_=hash_
-        )
+                i = i + 1
+                print('Ooops, wrong hash detected in try'
+                    ' #{rounds}. Recalculating pow... '.format(rounds= i)
+                )
```
Comment (todofixthis) [inline pow/ccurl_interface.py]: ❔ Should this use a logger?

File: test/test_ccurl_interface.py [hunk 2]
```diff
@@ -3,55 +3,318 @@
...
+                    )):
+            self.assertRaises(
+                ValueError,
+                ccurl_interface.attach_to_tangle,
+                # Enough if we have one tx in bundle
+                self.single_tx_bundle.as_tryte_strings(),
+                self.branch,
+                self.trunk,
+                mwm=14
+            )
```
Comment (todofixthis) [inline test/test_ccurl_interface.py]: Alternative:

```python
with self.assertRaises(ValueError):
    ccurl_interface.attach_to_tangle(...)
```

Whichever one you prefer 👍🏻

File: test/test_ccurl_interface.py [hunk 3]
```diff
@@ -3,55 +3,318 @@
...
+
+from six import PY2
+
+if PY2:
+    from mock import MagicMock, patch
+else:
+    from unittest.mock import MagicMock, patch
+
+# Calculate time in milliseconds (for timestamp)
+get_current_ms = lambda : int(round(time.time() * 1000))
```
Comment (todofixthis) [inline test/test_ccurl_interface.py]: 💭 Wondering if `get_current_ms` should always return a constant value during unit tests.  Depending on how deterministic the underlying pow process is, if we can control all of the inputs, we could make the unit tests even stricter when they scrutinise the pow'd trytes.

File: pow/ccurl_interface.py (hunk 1 above)
Comment (lzpap) [reply pow/ccurl_interface.py]: yep, good idea.

File: test/test_ccurl_interface.py (hunk 3 above)
Comment (lzpap) [reply test/test_ccurl_interface.py]: I decided to mock it away where it is needed :)

**Pull request merged**